tests/components/neato/test_config_flow.py

```diff
@@ -1,160 +1,156 @@
 
-async def test_user(hass, account):
-    """Test user config."""
-    flow = init_config_flow(hass)
+    assert len(hass.config_entries.async_entries(NEATO_DOMAIN)) == 1
+    assert len(mock_setup.mock_calls) == 1
 
-    result = await flow.async_step_user()
-    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
-    assert result["step_id"] == "user"
 
-    result = await flow.async_step_user(
-        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD, CONF_VENDOR: VENDOR_NEATO}
+async def test_abort_if_already_setup(hass: HomeAssistantType):
+    """Test we abort if Neato is already setup."""
+    entry = MockConfigEntry(
+        domain=NEATO_DOMAIN,
+        data={"client_id": "abcdef", "client_secret": "123456"},
+        version=2,
     )
+    entry.add_to_hass(hass)
 
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == USERNAME
-    assert result["data"][CONF_USERNAME] == USERNAME
-    assert result["data"][CONF_PASSWORD] == PASSWORD
-    assert result["data"][CONF_VENDOR] == VENDOR_NEATO
-
-    result = await flow.async_step_user(
-        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD, CONF_VENDOR: VENDOR_VORWERK}
+    # Should fail
+    result = await hass.config_entries.flow.async_init(
+        "neato", context={"source": config_entries.SOURCE_USER}
     )
-
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == USERNAME
-    assert result["data"][CONF_USERNAME] == USERNAME
-    assert result["data"][CONF_PASSWORD] == PASSWORD
-    assert result["data"][CONF_VENDOR] == VENDOR_VORWERK
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "already_configured"
 
 
-async def test_import(hass, account):
-    """Test import step."""
-    flow = init_config_flow(hass)
+async def test_reauth(hass: HomeAssistantType):
+    """Test initialization of the reauth flow."""
+    MockConfigEntry(
+        domain=NEATO_DOMAIN,
+        data={"username": "abcdef", "password": "123456", "vendor": "neato"},
+        version=1,
+    ).add_to_hass(hass)
 
-    result = await flow.async_step_import(
-        {CONF_USERNAME: USERNAME, CONF_PASSWORD: PASSWORD, CONF_VENDOR: VENDOR_NEATO}
+    # Should show form
+    result = await hass.config_entries.flow.async_init(
+        "neato", context={"source": config_entries.SOURCE_REAUTH}
     )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "reauth_confirm"
 
-    assert result["type"] == data_entry_flow.RESULT_TYPE_CREATE_ENTRY
-    assert result["title"] == f"{USERNAME} (from configuration)"
-    assert result["data"][CONF_USERNAME] == USERNAME
-    assert result["data"][CONF_PASSWORD] == PASSWORD
-    assert result["data"][CONF_VENDOR] == VENDOR_NEATO
+    # Form confirmed, init new flow
+    with patch(
+        "homeassistant.components.neato.config_flow.OAuth2FlowHandler.async_step_user",
```

Comment:
Don't patch the flow step. Assert that the existing entry is updated.

Reply:
Done